alleyinteractive/wp-newsletter-builder
# Issue-97: Remove dependency on Fieldmanager and manually create admin pages

### Summary

This PR aims to address the issue outlined in [#97](https://github.com/alleyinteractive/wp-newsletter-builder/issues/97), which involves removing the dependency on Fieldmanager for generating admin settings pages within WP Newsletter Builder and manually creating these settings pages instead. This change is intended to simplify the installation process for developers by eliminating the need for an additional plugin. Fixes #97

### Changes

- Remove Fieldmanager dependency for admin settings page creation.
- Manually create the admin settings pages within WP Newsletter Builder.

### Additional Information

During this process, consideration was given to utilizing Gutenberg for the admin edit page, as mentioned in the issue's description and supported by examples and suggestions in the issue's comments. Specifically, the use of the `useOption` hook from `alleyinteractive/alley-scripts` was proposed, along with the potential development of additional hooks or helpers for managing settings pages with Gutenberg.

### Test Instructions

1. Ensure WP Newsletter Builder is installed without Fieldmanager.
2. Navigate to the admin settings pages for WP Newsletter Builder.
3. Verify that all settings pages are accessible and function as expected without the Fieldmanager plugin.

### References

- Original Issue: [#97](https://github.com/alleyinteractive/wp-newsletter-builder/issues/97)
- Example of admin interface using Gutenberg: [Alley Developer Example](https://l.alley.dev/2cdafe0d6c)
- `useOption` hook suggestion: [useOption Hook](https://github.com/alleyinteractive/alley-scripts/tree/main/packages/block-editor-tools/src/hooks/use-option)
